frontend/src/Calendar.js

```diff
@@ -2,19 +2,108 @@ import React from 'react'
 import FullCalendar from '@fullcalendar/react'
 import dayGridPlugin from '@fullcalendar/daygrid'
 
+import moment from 'moment'
+import timeGridPlugin from '@fullcalendar/timegrid'
+
+import {fetchUserCalendar} from './script'
+
+// the conversion of one minute to 60000 milliseconds
+const MIN_TO_MS = 60000;
+
 export default class Calendar extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      fetchDone: false,
+      schedulesTimes: [],
+      scheduleIdx: 0     // Current schedule
+    }
+  }
+
+  componentDidMount(){
+    this.getAllSchedules()
+  }
+
+  async getAllSchedules() {
+    const schedules = await fetchUserCalendar()
+    for(const schedule of schedules) {
+      this.getMeetingTimes(schedule)
+    }
+    
+    this.setState({
+      fetchDone: true
+    })
+  }
+
+  getNextSchedule() {
+    if(this.state.scheduleIdx < this.state.schedulesTimes.length - 1) {
+      this.setState({scheduleIdx : this.state.scheduleIdx + 1})
+    }
+  }
+
+  getPrevSchedule() {
+    if(this.state.scheduleIdx > 0) {
+      this.setState({scheduleIdx : this.state.scheduleIdx - 1})
+    }
+  }
+
+  getMeetingTimes(schedule) {
+    let combinedMeetingTimes = []
+
+    for (const course of schedule.courses) {
+      // Lecture Sections
+      const lectureTime = this.getScheduleSections(course.name, course.lectureSection.meetingTimes)
+      // Lab Sections
+      if (course.labSection) {
+        const labTime = this.getScheduleSections(course.name, course.labSection.meetingTimes)
+        combinedMeetingTimes = combinedMeetingTimes.concat(labTime)
+      }
+      combinedMeetingTimes = combinedMeetingTimes.concat(lectureTime)
+    }
+
+    this.setState({
+      schedulesTimes:  [...this.state.schedulesTimes, combinedMeetingTimes]
```

Comment:
`schedulesTimes:  this.state.schedulesTimes.concat(combinedMeetingTimes)`

Reply:
I believe I tried using the `concat` method, but it did not work.

Reply:
Is `combinedMeetingTimes` an array? If yes, then that would explain it. If not, then `concat()` should return a new array containing everything in the first array, plus the element you gave it as a param

Reply:
The reason this matters is that `concat` behaves differently if the param you give it is an array.

```
const oneTwo = [1, 2];
const threeFour = [3, 4];
const three = 3;

oneTwo.concat(threeFour) => [1, 2, 3, 4]
[ ...oneTwo, threeFour ] => [1, 2, [3, 4]]

oneTwo.concat(three) => [1, 2, 3]
[ ...oneTwo, three ] = [1, 2, 3]
```

Reply:
Yes it is an array, so that explains it